backend/src/packages/users/user.repository.ts

```diff
@@ -16,7 +16,13 @@ class UserRepository implements Repository {
   public async findAll(): Promise<UserEntity[]> {
     const users = await this.userModel.query().execute();
 
-    return users.map((user) => UserEntity.initialize(user));
+    return users.map((user) =>
+      UserEntity.initialize({
+        ...user,
+        createdAt: new Date(user.createdAt),
```

Comment:
I've noticed you repeat this quite a few times. Maybe it is better to convert it into the date in the UserEntity constructor?

Reply:
They have Date types in the User Entity already, but without it I get TS types conflict errors.

Reply:
I mean you can pass it like a string to the constructor and then convert to date
```ts
{
    this.id = id;
    this.email = email;
    this.fullName = fullName;
    this.passwordHash = passwordHash;
    this.passwordSalt = passwordSalt;
    this.createdAt = new Date(createdAt);
    this.updatedAt = new Date(updatedAt);
  }
```

Reply:
so that you can simply pass user object to the `initialize` method

Reply:
I will have a conflict with the User Model anyway, because it is based on the Abstract, that had createdAt and updatedAt as type string

Reply:
Let's use @HIMDanny approach)

Reply:
So converting string to Date will be in one place

Reply:
Like I said above, I have User Model conflicts because of Abstract model. So can I change it?

Reply:
Finally done ;(

Reply:
Thank you very much for your patience))